libmamba/src/core/thread_utils.cpp

```diff
@@ -29,6 +29,25 @@ namespace mamba
 #ifndef _WIN32
     namespace
     {
+// `sigaddset` might be implemented as a macro calling `__sigbits(int)` function
+// At the same time `sigset_t` might be `unsigned int`
+// This causes compiler warning
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-conversion"
```

Comment:
I added comment explaining why this happens above
This doesn't happen in our code, but in the system code
But because `sigaddset` might be implemented as a macro, it's treated as our code.

Because, there is no bug, I completely disable this warning/error.
But I didn't want to disable it for the whole file, so I created a small function, where I applied this behaviour.